packages/oid4vci-holder/__tests__/shared/oid4vciHolderLogicAgentLogic.ts

```diff
@@ -0,0 +1,286 @@
+  })
+}
+
+export default (testContext: { getAgent: () => ConfiguredAgent; setup: () => Promise<boolean>; tearDown: () => Promise<boolean> }): void => {
+  describe('OID4VI Holder Agent Plugin', (): void => {
+    let agent: ConfiguredAgent
+
+    beforeAll(async (): Promise<void> => {
+      await testContext.setup()
+      agent = testContext.getAgent()
+    })
+
+    afterAll(testContext.tearDown)
+
+    it('should get initialization data using pre-authorized_code and openid-initiate-issuance draft < 9', async (): Promise<void> => {
+      succeedWithAFullFlowWithClientSetup()
+      await expect(
+        agent.oid4vciHolderGetInitiationData({
+          requestData: {
+            uri: INITIATE_QR_PRE_AUTHORIZED,
+          },
+        }),
+      ).resolves.toEqual(GET_INITIATION_DATA_PRE_AUTHORIZED_OPENID_INITIATE_ISSUANCE)
+    })
+
+    it('should get initialization data using pre-authorized_code and draft 11 >', async (): Promise<void> => {
+      succeedWithAFullFlowWithClientSetup()
+      await expect(
+        agent.oid4vciHolderGetInitiationData({
+          requestData: {
+            uri: OFFER_QR_PRE_AUTHORIZED,
+          },
+        }),
+      ).resolves.toEqual(GET_INITIATION_DATA_PRE_AUTHORIZED_OPENID_CREDENTIAL_OFFER)
+    })
+
+    it('should get initialization data using pre-authorized_code and https draft < 9', async (): Promise<void> => {
+      succeedWithAFullFlowWithClientSetup()
+      await expect(
+        agent.oid4vciHolderGetInitiationData({
+          requestData: {
+            uri: HTTPS_INITIATE_QR,
+          },
+        }),
+      ).resolves.toEqual(GET_INITIATION_DATA_PRE_AUTHORIZED_HTTPS_INITIATE_FLOW)
+    })
+
+    it('should get initialization data using authorization_code and https draft 11 >', async (): Promise<void> => {
+      succeedWithAFullFlowWithClientSetup()
+      await expect(
+        agent.oid4vciHolderGetInitiationData({
+          requestData: {
+            uri: HTTPS_OFFER_QR_AUTHORIZATION_CODE,
+          },
+        }),
+      ).resolves.toEqual(GET_INITIATION_DATA_AUTHORIZATION_CODE_HTTPS)
+    })
+
+    it('should get initialization data using pre-authorized_code and https draft 11 >', async (): Promise<void> => {
+      succeedWithAFullFlowWithClientSetup()
+      await expect(
+        agent.oid4vciHolderGetInitiationData({
+          requestData: {
+            uri: HTTPS_OFFER_QR_PRE_AUTHORIZED,
+          },
+        }),
+      ).resolves.toEqual(GET_INITIATION_DATA_PRE_AUTHORIZED_CODE_HTTPS)
+    })
+
+    it('should initialize the OID4VCI Client in the state machine without errors', async () => {
+      succeedWithAFullFlowWithClientSetup()
+      const linkHandler = new OID4VCIHolderLinkHandler({
+        protocols: [
+          new RegExp('http:\\/\\/.*\\?.*credential_offer=.+'),
+          new RegExp('http:\\/\\/.*\\?.*credential_offer_uri=.+'),
+          new RegExp('https:\\/\\/.*\\?.*credential_offer=.+'),
```

Comment:
What is going one with 4 regexes. If you wanted to these could be one, but for sure combine the http with https. It is literally appending 's?' to the http version to also support https

Reply:
Also it would be (_uri)? to support both the by value and by reference versions of the offer

Reply:
I'm using the existing implementation from the cloud wallet to make sure I'm testing our actual use case.

Reply:
Already changed it in integration branches, as this makes little sense